### ISSUE TITLE
Automatically detect and set CP932 encoding (as different from SJIS)

### ISSUE THREAD

**Issue by Kimik0** [fontforge/fontforge#2831]
As discussed by @jtanx in #2820, CP932 encoded fonts are detected/read as SJIS encoded. Please make it so that FontForge can automatically detect and set the correct encoding. See over there for @jtanx's ideas on how to go about this and some fonts to test with.

IANAP so I'll leave the rest up to y'all ^_^;

('T would be groovy if this could be worked into other font handling libraries so it's not even necessary to reencode such fonts for daily use.)

**Comment by JoesCat** [comment]
Adding link for more details, http://www8.plala.or.jp/tkubota1/unicode-symbols-map2.html

**Comment by Kimik0** [comment]
Another thing, some of these fonts that I can take a closer look at now that #2820 is fixed, seem to be different extensions of Shift-JIS. When I try the "add encoding name", "force encoding" method with either CP932 or Windows-31J encoding, there are always a bunch of characters left over at the end, some copies of others, some unique. I think they might be one or more of the encodings listed at https://github.com/adobe-type-tools/cmap-resources/tree/master/cmapresources_japan1-6/CMap but I can't figure out which because they're not available in iconv.

**Comment by jtanx** [comment]
Yeah in all likelihood they're using extensions based on ShiftJIS. That makes it very difficult to automatically detect what encoding it's actually using and to apply the correct one.

**Comment by Kimik0** [comment]
It still would be good to have those encodings listed in FontForge so you can try out which one fits best.
